dhall/src/Dhall/ETags.hs

```diff
@@ -0,0 +1,256 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE LambdaCase #-}
+
+-- | This module contains the implementation of the @dhall tags@ command
+
+module Dhall.ETags
+    ( -- * ETags
+      generate
+    ) where
+
+import Control.Exception (handle, SomeException(..))
+import Control.Monad (foldM)
+import Data.List (isSuffixOf)
+import Data.Maybe (fromMaybe)
+import Data.Semigroup (Semigroup(..))
+import Dhall.Map (foldMapWithKey)
+import Data.Text (Text)
+import Data.Text.Encoding (encodeUtf8)
+import Dhall.Util (Input(..))
+import Dhall.Core (Expr(..), Binding(..))
+import Dhall.Src (Src(srcStart))
+import Dhall.Parser (exprFromText)
+import System.Directory ( doesDirectoryExist
+#if MIN_VERSION_directory(1,3,0)
+                        , pathIsSymbolicLink
+#elif MIN_VERSION_directory(1,2,6)
+                        , isSymbolicLink
+#endif
```

Comment:
If you use a qualified import, you can remove this CPP.

Reply:
But there will be warning. Is it okay?

Reply:
Not sure what you mean. If you just `import qualified System.Directory` there shouldn't be any warning. We use `getDirectoryContents` in any case.

Reply:
Oh, yes, I see.

Reply:
Fixed.